client/req.go

```diff
@@ -39,19 +39,23 @@ func (c *Conn) writeCommandStr(command byte, arg string) error {
 func (c *Conn) writeCommandUint32(command byte, arg uint32) error {
 	c.ResetSequence()
 
-	return c.WritePacket([]byte{
-		0x05, //5 bytes long
-		0x00,
-		0x00,
-		0x00, //sequence
+	buf := utils.ByteSliceGet(9)
```

Comment:
re-use of byte slices and byte buffers was done on the packet reading path but not so much on the packet writing path. Our application is write heavy and as a result we saw lots of allocations when writing. Some of the changes you'll see in this PR are to re-use byte slices and buffers where it makes sense.